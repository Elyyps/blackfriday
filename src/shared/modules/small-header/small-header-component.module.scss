.small-header {
  color: $global-inverse-color;
  text-align: center;
  position: relative;
  z-index: 1;
  margin-bottom: 40px;
  padding: 72px 0;
  @media (max-width: $breakpoint-small-max) {
    margin-bottom: 24px;
    padding: 40px 0;
  }
  @media (max-width: $breakpoint-xsmall-max) {
    text-align: left;
  }
  &::before,
  &::after {
    position: absolute;
    left: 0;
    top: 0;
    right: 0;
    bottom: 0;
    content: "";
    background: $gray-dark;
    z-index: -1;
  }
  &::after {
    background: url("../../../assets/pattern.svg");
    background-position: 50% 0;
    background-size: 2000px auto;
    opacity: 0.5;
  }

  h1 {
    font-weight: bold;
    color: $global-inverse-color;
  }
}
