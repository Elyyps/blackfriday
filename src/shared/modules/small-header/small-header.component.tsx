import * as React from "react";
import styles from "./small-header-component.module.scss";

export interface ISmallHeaderComponentProps {
  backButtonProps?: {};
  breadcrumbItems?: {}[];
  title?: string;
}

const SmallHeaderComponent = (props: ISmallHeaderComponentProps) => (
  <div className={styles["small-header"]}>
    <div className="uk-container">
      Hier komt de breadcrumb
      {props.title && <h1>{props.title}</h1>}
    </div>
  </div>
);

export { SmallHeaderComponent };
