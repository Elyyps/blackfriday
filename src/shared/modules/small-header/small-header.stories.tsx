import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { SmallHeaderComponent } from "./small-header.component";
import { withProvider } from "@app/util";
import { headerContent } from "@app/api/core/header-content";

storiesOf("Small Header", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <SmallHeaderComponent headerContent={headerContent} />);
