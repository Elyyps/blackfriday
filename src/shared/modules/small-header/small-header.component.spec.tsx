import React from "react";
import { shallow } from "enzyme";
import { SmallHeaderComponent } from "./small-header.component";

describe("[SmallHeader]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<SmallHeaderComponent title="Black Friday 2019 blog" />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
