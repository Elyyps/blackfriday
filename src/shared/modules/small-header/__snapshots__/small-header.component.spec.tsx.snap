// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SmallHeader] should render component without crashing 1`] = `
<div
  className="small-header"
>
  <div
    className="uk-container"
  >
    <HeaderContentComponent
      breadcrumbProps={
        Object {
          "backButton": Object {
            "title": "Terug",
            "url": "/#",
          },
          "links": Array [
            Object {
              "title": "Home",
              "url": "/#",
            },
            Object {
              "title": "Blog",
              "url": "/#",
            },
          ],
        }
      }
      headerMobilePosition="left"
      subtitle="Deelnemende winkels van 2019"
      text="Bodytext voor extra informatie wanneer nodig"
      title="Black Friday Winkels"
    />
  </div>
</div>
`;
